main (1).cpp

```diff
@@ -0,0 +1,48 @@
+#include <iostream>
+#include <cmath>
+/**
+*\ brief recursive calculation of the sum of the first n terms of the sequence
+*\ param n - number of sequence members
+*\ return the amount
+*/
+double sequenceRec(int n);
+/**
+*\ brief recursive calculation of the sum of the first terms of a sequence not less than a given number e
+*\ param e - the specified number
+*\ return the amount
+*/
+double sequenceRec2(double e);
+int main()
+{
+int n;
+double e;
+std::cout << "Enter n "; std::cin >> n;
+std::cout << "Enter e "; std::cin >> e;
+std::cout << "The sum of the members of the sequence for n = " << n << " " << sequenceRec(n) << "\n";
+std::cout << "The sum of the members of the sequence for e = " << e << " " << sequenceRec2(e) << "\n";
+return 0;
+}
+double sequenceRec(int n)
```

Comment:
Если параметры функции не изменяются, то они явно должны быть обозначены `const`
[By Alex Allain](https://www.cprogramming.com/tutorial/const_correctness.html)
>The const keyword allows you to specify whether or not a variable is modifiable. You can use const to prevent modifications to variables and const pointers and const references prevent changing the data pointed to (or referenced).
>Const дает вам возможность более четко документировать вашу программу и фактически обеспечивать соблюдение этой документации. Ключевое слово const обеспечивает вашим пользователям гарантии, позволяющие оптимизировать производительность без угрозы повреждения их данных, обеспечивая соблюдение вашей документации. Например, константные ссылки позволяют указать, что упомянутые данные не будут изменены; это означает, что вы можете использовать константные ссылки как простой и немедленный способ повышения производительности для любой функции, которая в настоящее время принимает объекты по значению, не беспокоясь о том, что ваша функция может изменить данные. Даже если это произойдет, компилятор предотвратит компиляцию кода и предупредит вас о проблеме. С другой стороны, если бы вы не использовали константные ссылки, у вас не было бы простого способа гарантировать, что ваши данные не были изменены.